### PR TITLE
open all task folders in one click

### PR DESCRIPTION
**Pull request by juneluo**
### Description
Fixes bug priority low 7 of PROJECTS/TASKS/WBS COMPONENT:
Make clicking “All” on a WBS open all folders to show all tasks

### Mainly changes explained:
Load all tasks of the WBS when open the WBS task page but don't show the sub tasks in the initial state. Open all folders when the filterstate is set to 'all' by clicking the 'all' button.
https://user-images.githubusercontent.com/9314962/215760671-45b37427-eb5e-4f02-a904-dcc90c51e969.mov

### How to test:
check into current branch
do npm install and ... to run this PR locally
Logged in as Amin → Other Links → Projects → AAA USE THIS PROJECT → Click WBS Icon → Choose a WBS → Click “all”
img

### CODE REVIEW

**Comment by RafaelCastilhoCastro** [comment]
Hi @juneluo,
I tested the PR and it didn't work as intended.
If I undestood properly, when opening the WBS section the sub tasks shouldn't appear unless the user click "all", yes?
In my tests the sub tasks are being loaded when the page loads. (Please check video)

https://user-images.githubusercontent.com/65736872/215831931-4f7b8717-bbf6-42c1-b7a5-d1869dab95a2.mp4

**Comment by Gukal7** [comment]
Hi @juneluo, I tested your code and it is working fine but like you said it takes 10+ seconds to finish loading and then only we can use the All button properly. In my opinion it makes things confusing to the user who opens the WBS and the tasks/subtasks keeps on coming and going for 10+ seconds and then everything stabilizes. 

https://user-images.githubusercontent.com/30505274/215910739-632bc048-7f3b-4a46-a82e-be696f7328d2.mov

**Comment by AnishPandita** [comment]
Hi @juneluo I tested your PR and it is working correctly. The subtasks are visible after clicking the all button. But as @Gukal7 suggested it would be really beneficial if the loading time could be reduced. That will provide a more seamless user experience.

https://user-images.githubusercontent.com/111321945/215930676-8cdae09b-a113-4f8b-b7b3-d08438c3bd9b.mov

 The code looks clear and according to standards.

**Comment by tingedudu8** [comment]
Hi @juneluo, is it possible to delete the process to load all the tasks including the sub tasks? Since while the load time is reduced, the user might still thought it was the internet problem. It would be much better to add a small icon of "loading" animation or simply delete it.

**Comment by RafaelCastilhoCastro** [comment]
Hi @juneluo,
I performed new tests and the loading time really improved but as noted before it doesn't look good since the page is shifting and loading long after being rendered for the user.

In my tests the process followed these steps:
1. The page is rendered displaying only the main tasks
2. After 3-4 seconds the page displays a few subtasks
3. After 2-3 seconds all subtasks as displayed
4. After 1-2 seconds the subtasks are hidden, only main tasks remain.

Maybe some process are overlapping or repeting themselves causing these effects.

File: src/components/Projects/WBS/WBSDetail/WBSTasks.jsx [hunk 1]
```diff
@@ -29,21 +29,51 @@ const WBSTasks = props => {
   const [isShowImport, setIsShowImport] = useState(false);
   const [selectedId, setSelectedId] = useState(null);
   const [filterState, setFilterState] = useState('all');
+  const [openAll, setOpenAll] = useState(false);
+
+  const load = async()=>{
+    const le = [0,1,2,3,4];
```
Comment (RafaelCastilhoCastro) [inline src/components/Projects/WBS/WBSDetail/WBSTasks.jsx:35]: When possible please rename the variables as to improve their readability and funcionality. Thank you.

**Review by PrabhjotSembhi — COMMENTED** [review]
The code is good. As Rafael renaming variables will be good. 
I also prefer destructuring the props ( Thats my personal opinion)

**Review by PrabhjotSembhi — COMMENTED** [review]
The "All" button is working correctly but there is a bug.
If you wait for Some sub-tasks appear for 3-4 seconds. 
I have recorded a video you can check and understand in more detail

https://user-images.githubusercontent.com/82662768/216533613-605be65b-7dca-4ee8-9ef3-282649b9a9fa.mp4

**Comment by Filipe-Oliv** [comment]
Hi @juneluo 

I have tested your PR: 
It is working properly like in your last video. 
The subtasks are available after clicking the all button.

[week 03 - 02-07-23 (5).webm](https://user-images.githubusercontent.com/114119668/217357951-84b424a8-5f73-4226-9b21-8e81ba9fd7bf.webm)

**Comment by NyahJ** [comment]
Hi @juneluo

Thanks for all the wonderful work you have done!
I have tested your PR, functions are clear and complete, while there was a little issue when "All" was clicked:

The data should end after # 18 Task, while there was a long list of the additional and duplicated data(in the red rectangle) following, could you pls take a look at this?
截屏2023-02-07 下午5 21 09

**Comment by Filipe-Oliv** [comment]
Hi @juneluo 

I've looked at the @NyahJ comment and it's working properly.

[week 03 - 02-08-23 (1).webm](https://user-images.githubusercontent.com/114119668/217604264-2d3a8b3a-585d-4773-b513-fa24cc301daa.webm)

**Review by tingedudu8 — COMMENTED** [review]
Hi @juneluo, reviewed and the loading button makes the whole process much clear now, and the sorting problem has been fixed, thanks!

https://user-images.githubusercontent.com/70128933/217655201-a2af48bf-c52b-436a-bd7e-ba9ea94ea8fd.mp4

**Comment by one-community** [comment]
If no problems, someone please approve this PR and I'll move it to the Management Review Team

**Review by tingedudu8 — APPROVED** [review]

**Comment by rauleffting** [comment]
Hi @juneluo. I tested your PR and it's working as intended. 

proof

Ready to be merged!

**Comment by one-community** [comment]
Excellent, thank you everyone. Merged :)

**Pull request merged**